cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1538,6 +1576,14 @@ func generateMountsModeRun(mst *initramfsMountsState) error {
 		}
 	}
 
+	// Setup bind-mounts in /sysroot
+	if err := doBaseBindMount(); err != nil {
```

Comment:
this needs unit tests for the case where the mount unit files don't exist, let me know if you need help writing these tests, probably you wanna start with copying and modifying these 2 tests:
* `TestInitramfsMountsRunModeWithSaveHappyRealSystemdMount`
* `TestInitramfsMountsRunModeNoSaveUnencryptedHappy`

and have the start of those tests delete the mount unit files that were mocked in the `SetUpTest()`. You will then see some failures about the expected systemctl calls and probably just need to add these new calls in.

Reply:
ah I see you added some tests in https://github.com/snapcore/snapd/pull/10351, typically we prefer to keep the tests in the same PR as the one that adds the features, so if you could just move those tests to this PR that would simplify reviewing this  PR for others. 

Also I see that in this PR you mock the mount unit files existing and don't modify the tests while the other PR has tests about when the mount units are missing - we need both kinds of tests hence my suggestion to add specific tests which check that things work when the mount units are missing, but since actually that situation will become "legacy" support, maybe we ought to flip it and modify all the tests the way you did in #10351, then take the tests I mentioned above and mock add the mount unit files and those two tests can be for the legacy situation. Does that make sense?